src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php

```diff
@@ -579,14 +602,24 @@ public function setMainAccount(Contact $contact, $data)
                 $accountContact->setPosition($position);
             } else {
                 // else create new one
-                $this->createMainAccountContact($contact, $account, $position);
+                $accountContact = $this->createMainAccountContact($contact, $account, $position);
+                $this->domainEventCollector->collect(new AccountContactAddedEvent($accountContact));
             }
         } else {
             // if a main account exists - remove it
             if ($accountContact = $this->getMainAccountContact($contact)) {
                 // if this contact is the main-Contact - set mainContact to null
                 if ($accountContact->getAccount()->getMainContact() === $contact) {
                     $accountContact->getAccount()->setMainContact(null);
+                    $this->domainEventCollector->collect(
+                        new AccountContactRemovedEvent(
```

Comment:
would prefer to pass the `account` and the `contact` entity here to be consistent with the `AccountMediaRemovedEvent`